tests/v2/validation/rbac/rbac_psa_test.go

```diff
@@ -0,0 +1,336 @@
+		require.Error(rb.T(), err)
+		errMessage := strings.Split(err.Error(), ":")[0]
+		assert.Equal(rb.T(), "Resource type [namespace] is not updatable", errMessage)
+	case roleProjectOwner, roleProjectMember:
+		require.Error(rb.T(), err)
+		errStatus := strings.Split(err.Error(), ".")[1]
+		rgx := regexp.MustCompile(`\[(.*?)\]`)
+		errorMsg := rgx.FindStringSubmatch(errStatus)
+		assert.Equal(rb.T(), "403 Forbidden", errorMsg[1])
+
+	}
+
+	rb.T().Logf("Validate deletion of the PSA labels as %v", role)
+
+	for label := range labels {
+		if strings.Contains(label, psaWarn) || strings.Contains(label, psaAudit) || strings.Contains(label, psaEnforce) {
+			delete(labels, label)
+		}
+	}
+
+	deleteLabelsNS := rb.ConvertNS(rb.adminNamespace)
+	deleteLabelsNS.Labels = labels
+
+	_, err = rb.steveNonAdminClient.SteveType(namespaces.NamespaceSteveType).Update(rb.adminNamespace, deleteLabelsNS)
+	switch role {
+	case restrictedAdmin, roleOwner:
+		require.NoError(rb.T(), err)
+		expectedLabels := getPSALabels(response, labels)
+		assert.Equal(rb.T(), 0, len(expectedLabels))
+		_, err = createDeployAndWait(rb.steveNonAdminClient, rb.client, rb.cluster.ID, containerName, containerImage, rb.adminNamespace.Name)
+		require.NoError(rb.T(), err)
+	case roleMember, roleProjectReadOnly:
+		require.Error(rb.T(), err)
+		errMessage := strings.Split(err.Error(), ":")[0]
+		assert.Equal(rb.T(), "Resource type [namespace] is not updatable", errMessage)
+	case  roleProjectOwner, roleProjectMember:
+		errStatus := strings.Split(err.Error(), ".")[1]
+		rgx := regexp.MustCompile(`\[(.*?)\]`)
+		errorMsg := rgx.FindStringSubmatch(errStatus)
+		assert.Equal(rb.T(), "403 Forbidden", errorMsg[1])
+
+	}
+
+	rb.T().Logf("Validate creation of new namespace with PSA labels as %v", role)
+
+	labels = map[string]string{
+		psaWarn:    pssBaselinePolicy,
+		psaEnforce: pssBaselinePolicy,
+		psaAudit:   pssBaselinePolicy,
+	}
+	namespaceName := namegenerator.AppendRandomString("testns-")
+	namespaceCreate, err := namespaces.CreateNamespace(rb.nonAdminUserClient, namespaceName, "{}", labels, map[string]string{}, rb.adminProject)
+
+	switch role {
+	case restrictedAdmin, roleOwner:
+		require.NoError(rb.T(), err)
+		expectedLabels := getPSALabels(response, labels)
+		assert.Equal(rb.T(), labels, expectedLabels)
+		_, err = createDeployAndWait(rb.steveNonAdminClient, rb.client, rb.cluster.ID, "nginx", "nginx", namespaceCreate.Name)
+		require.NoError(rb.T(), err)
+	case roleMember, roleProjectOwner, roleProjectMember, roleProjectReadOnly:
+		require.Error(rb.T(), err)
+		if role == roleMember || role == roleProjectReadOnly {
+			errMessage := strings.Split(err.Error(), ":")[0]
+			assert.Equal(rb.T(), "Resource type [namespace] is not creatable", errMessage)
+		} else {
+			errStatus := strings.Split(err.Error(), ".")[1]
+			rgx := regexp.MustCompile(`\[(.*?)\]`)
+			errorMsg := rgx.FindStringSubmatch(errStatus)
+			assert.Equal(rb.T(), "403 Forbidden", errorMsg[1])
+		}
+	}
```

Comment:
I've got the same question from the previous PR for this if/else statement grouping inside a switch/case: https://github.com/rancher/rancher/pull/40157#discussion_r1104566824